feat: flatpaksync submodule

lucasvsr opened the pull request:
# `flatpaksync` submodule

The `flatpaksync` submodule can be used to synchronize user-installed flatpaks into a gist or traditional repository.

To activate it, simply follow the example below:

## Example configuration

```yaml
- type: bling
  # ...other submodules...
  - flatpaksync
```

Once the submodule is activated, users can create a file `$HOME/.config/flatpaksync/env` informing the repository that will be used to synchronize their apps in the POSIX standard:

```bash
GIST_REPO=<YOUR_REPO>
```

If the user has not yet installed their flatpaks, has already done the step above and has a `flatpak.list` file in the repository, simply use the `flatpakcheckout` binary to perform the installation and start the synchronization.

**It is important to note that this submodule will NOT enable Flathub. If your applications come from there, you will need to enable Flathub before running it.**

If the user has already configured their repository in the `$HOME/.config/flatpaksync/env` file but already has their flatpaks installed, simply create the `$HOME/.config/flatpaks.user.installed` file to start the synchronization.

lucasvsr committed:
feat: flatpaksync module

lucasvsr committed:
chore: translations

lucasvsr committed:
fix: source env file

lucasvsr committed:
Update README.md

lucasvsr committed:
chore: dirs

lucasvsr committed:
chore: dirs

lucasvsr committed:
chore: dirs

xynydev commented:
Will look closer at this and the other flatpak-related PR probably tomorrow, but
- Why just gist, why not just a raw URL to a file that could be anywhere?
- Since this takes in no recipe.yml configuration, shouldn't this be inside the bling module and not a module unto itself?

lucasvsr commented:
> Will look closer at this and the other flatpak-related PR probably tomorrow, but
> 
> * Why just gist, why not just a raw URL to a file that could be anywhere?
> * Since this takes in no recipe.yml configuration, shouldn't this be inside the bling module and not a module unto itself?

Okay, I'm waiting for the review.

About the questions:

- It needs to be a repository, actually. I use a gist because I think it's simpler, but there's nothing stopping it. So, when a flatpak is installed, the repository will be checked out and the `flatpak.list` file will be updated (or created, if it doesn't already exist).

- Good idea, I didn't think of that. I will make the adjustment and let you know as soon as I commit.

lucasvsr committed:
chore: including functionality as a submodule of the bling module

lucasvsr commented:
Done, @xynydev, refactoring available for review.

lucasvsr committed:
fix: rm at sync.

xynydev committed:
docs: chore: include submodule documentation in the main bling README

There's a lot of bling "submodules" that might require documentation. Including them inside the same README is beneficial, because it makes the appear on the ublue website too, and their documentation is often not that long.

xynydev committed:
chore: delete special flatpaksync documentation file

xynydev commented:
With [6fdfffc](https://github.com/ublue-os/bling/pull/62/commits/6fdfffcb69f1aee173c171cc57e1aa5eed5a184b) and [462a12f](https://github.com/ublue-os/bling/pull/62/commits/462a12f5ad08cee69d3bff4f876449c4e8340b7e) I moved the documentation for `flatpaksync` to the bling module's README such that it shows up on the ublue website.

xynydev committed:
docs: chore: remove mention of flatpaksync.md, which doesn't exist

xynydev commented on modules/bling/README.md:
````diff
@@ -19,4 +19,22 @@ install:
+Once the submodule is activated, users can create a file `$HOME/.config/flatpaksync/env` informing the repository that will be used to synchronize their apps in the POSIX standard:
+
+```bash
+GIST_REPO=<YOUR_REPO>
````
Overall, this documentation leaves the purpose of this module unspecified. Does this module upload a list of flatpaks installed on your system to a Github Gist (a proprietary Github product), or is any `git` repository supported? There should be examples of what to set `GIST_REPO` to. Or, does the sync happen in two directions, where editing the file in the gist changes the flatpaks on your system as well? What is the structure of the repository? Is the repository cloned into `~/.config/flatpaksync/` or where exactly?

And a big question; What is the usecase where this approach has benefits over the `default-flatpaks` module? 

Reading the documentation multiple times and reading the source code as well does give me answer to these questions, which I had while reading the docs for the first time. It's already pretty good as of now, but needs a bit of polish. And if you're more comfortable with someone else writing the documentation, feel free to just explain stuff as a reply and I'll transform that into docs on a later date.

lucasvsr replied:
It can be any git repository, actually. In my case, I used a gist because I found it simpler. Synchronization is bidirectional, every time the user installs a flatpak using the `--user` clause, the `flatpak.list` file will be updated. Maybe changing the variable name would help...

The repository is cloned to `/tmp/sync`. It is not necessary to set up any structure in the repository, the script will take care of creating the `flatpak.list` file in the root of the repository, if necessary, or updating it.

I didn't make this clear in the documentation because I wanted to make this timing transparent. Of course, any user with a little more experience will be able to understand everything I say by looking at the `flatpakcheckout` and `flatpaksync` scripts.

From what I understand from the `default-flatpaks` module, it will enable repositories and install apps, either in the system context or in the user context, but it will not synchronize the apps installed by the user over time. This submodule does not have the same function, using it the user can synchronize their flatpaks, which will not always be the same as the ones other PC users use, and facilitate their installation if they format the computer, for example.

Please feel free to rewrite the documentation as you see fit.

xynydev replied:
Thanks for the quick and clear response! Maybe the variable should be named `GIT_REPO` then. I'm gone for today but I'll do some more changes as I see fit and merge the PR soon.

Commit:
chore: rename GIST_REPO to GIT_REPO

Commit:
docs: detail & misc changes to flatpaksync docs

xynydev commented:
Does the new documentation look good? I'm ready to merge whenever. Actually, it'll take a while for `ublue-os/website` to even pick this up, so I'll just merge, and we'll change the docs if there's errors, alright,
